packages/medusa/src/utils/product-category/index.ts

```diff
@@ -0,0 +1,40 @@
+import { FindOptionsWhere } from "typeorm"
+import { ProductCategory } from "../../models"
+import { isDefined } from "medusa-core-utils"
+
+export const categoryMatchesScope = (
+  category: ProductCategory,
+  query: FindOptionsWhere<ProductCategory>
+): boolean => {
+  const scopeActive = isDefined(query.is_active)
+  const scopePublic = isDefined(query.is_internal)
```

Comment:
thought: Wouldn't `query.is_internal` mean, that they are not publicly scoped?
```suggestion
  const scopePublic = isDefined(query.is_internal)
```

Reply:
todo: I believe we should not only check for a defined value but also that it is `false`

Reply:
If this is the case, this should be reflected in the entire method body.

Reply:
ooh, good catch.

Reply:
oh wait, thats already happening. I got confused for a sec there. I've removed the confusing variable names. We're basically checking if any of those parameters are defined, and based on that, do a scoping.

Reply:
Got it 👍